ui/pages/notifications/notification-components/erc1155-sent-received/erc1155-sent-received.tsx

```diff
@@ -191,16 +191,15 @@ export const components: NotificationComponent<ERC1155Notification> = {
       const { nativeBlockExplorerUrl } = getNetworkDetailsByChainId(
         `0x${chainId}` as keyof typeof CHAIN_IDS,
       );
+      if (!nativeBlockExplorerUrl) {
+        return null;
+      }
       return (
         <NotificationDetailButton
           notification={notification}
           variant={ButtonVariant.Secondary}
           text={t('notificationItemCheckBlockExplorer') || ''}
-          href={
-            nativeBlockExplorerUrl
-              ? `${nativeBlockExplorerUrl}//tx/${notification.tx_hash}`
-              : '#'
-          }
+          href={`${nativeBlockExplorerUrl}/tx/${notification.tx_hash}`}
```

Comment:
I just realized that the way the block explorer links are created within this PR doesn't account for custom explorer links. We need to use the `blockExplorerUrl` that comes from the object returned by `getRpcPrefsForCurrentProvider`

Also, `getBlockExplorerLink` can be used to construct the url.

Reply:
Thank you @danjm . I'm a bit confused :)

If I'm not mistaken, `getRpcPrefsForCurrentProvider` returns the `blockExplorerUrl` based on the current provider. However, in our case, the notifications are about the events that happened for a particular account on any chain it has interacted with. In this sense, a notification could concern an action that occurred on chain X while my wallet is set to use chain Y. That's why I created this method `getNetworkDetailsByChainId` here `ui/helpers/utils/notification.util.ts`.

The idea is that by passing the chainID, I get the necessary data for that chain ID:

```javascript
nativeCurrencyName,
nativeCurrencySymbol,
nativeCurrencyLogo,
nativeCurrencyAddress,
nativeBlockExplorerUrl,
```

I don't quite understand the implementation of `getBlockExplorerLink`, but I see these other methods:

```javascript
interface TransactionInterface {
    hash: string;
    chainId: string;
    metamaskNetworkId: string;
}

interface RpcPrefsInterface {
    blockExplorerUrl?: string;
}

export declare function createCustomExplorerLink(hash: string, customNetworkUrl: string): string;
export declare function createExplorerLink(hash: string, network: string): string;
export declare function createExplorerLinkForChain(hash: string, chainId: string): string;
export declare function getBlockExplorerLink(transaction: TransactionInterface, rpcPrefs?: RpcPrefsInterface): string;
```

Perhaps I can use `createExplorerLinkForChain`? I just don't see it used anywhere in the Extension

Reply:
Okay, you are right, `getRpcPrefsForCurrentProvider` won't work.

We still need to get the user's preferred block explorer url. We may need a new selector for this. You could use `getAllNetworks` to get a list of all networks, then find the one that matches the current networks id, and then get the blockexplorer URL for that.

Otherwise we will default to showing the user the information on etherscan, even if they explicitly configured their network to use a block explorer other than etherscan.

Reply:
@danjm sorry, I didn't understand: is there a way for the user to set the preferred block explorer for each chain?

And can I ask if it is correct that the source to find the explorer is this?

```javascript
/**
 * Map of all Etherscan supported networks.
 */
export const ETHERSCAN_SUPPORTED_NETWORKS = {
  [CHAIN_IDS.GOERLI]: {
    domain: defaultEtherscanDomain,
    subdomain: `${defaultEtherscanSubdomainPrefix}-${
      CHAIN_ID_TO_TYPE_MAP[CHAIN_IDS.GOERLI]
    }`,
  },
  [CHAIN_IDS.MAINNET]: {
    domain: defaultEtherscanDomain,
    subdomain: defaultEtherscanSubdomainPrefix,
  },
  [CHAIN_IDS.SEPOLIA]: {
    domain: defaultEtherscanDomain,
    subdomain: `${defaultEtherscanSubdomainPrefix}-${
      CHAIN_ID_TO_TYPE_MAP[CHAIN_IDS.SEPOLIA]
    }`,
  },
  [CHAIN_IDS.LINEA_GOERLI]: {
    domain: 'lineascan.build',
    subdomain: 'goerli',
  },
  [CHAIN_IDS.LINEA_SEPOLIA]: {
    domain: 'lineascan.build',
    subdomain: 'sepolia',
  },
  [CHAIN_IDS.LINEA_MAINNET]: {
    domain: 'lineascan.build',
    subdomain: defaultEtherscanSubdomainPrefix,
  },
  [CHAIN_IDS.BSC]: {
    domain: 'bscscan.com',
    subdomain: defaultEtherscanSubdomainPrefix,
  },
  [CHAIN_IDS.BSC_TESTNET]: {
    domain: 'bscscan.com',
    subdomain: `${defaultEtherscanSubdomainPrefix}-testnet`,
  },
  [CHAIN_IDS.OPTIMISM]: {
    domain: defaultEtherscanDomain,
    subdomain: `${defaultEtherscanSubdomainPrefix}-optimistic`,
  },
  [CHAIN_IDS.OPTIMISM_TESTNET]: {
    domain: defaultEtherscanDomain,
    subdomain: `${defaultEtherscanSubdomainPrefix}-sepolia-optimistic`,
  },
  [CHAIN_IDS.POLYGON]: {
    domain: 'polygonscan.com',
    subdomain: defaultEtherscanSubdomainPrefix,
  },
  [CHAIN_IDS.POLYGON_TESTNET]: {
    domain: 'polygonscan.com',
    subdomain: `${defaultEtherscanSubdomainPrefix}-mumbai`,
  },
  [CHAIN_IDS.AVALANCHE]: {
    domain: 'snowtrace.io',
    subdomain: defaultEtherscanSubdomainPrefix,
  },
  [CHAIN_IDS.AVALANCHE_TESTNET]: {
    domain: 'snowtrace.io',
    subdomain: `${defaultEtherscanSubdomainPrefix}-testnet`,
  },
  [CHAIN_IDS.FANTOM]: {
    domain: 'ftmscan.com',
    subdomain: defaultEtherscanSubdomainPrefix,
  },
  [CHAIN_IDS.FANTOM_TESTNET]: {
    domain: 'ftmscan.com',
    subdomain: `${defaultEtherscanSubdomainPrefix}-testnet`,
  },
  [CHAIN_IDS.MOONBEAM]: {
    domain: 'moonscan.io',
    subdomain: `${defaultEtherscanSubdomainPrefix}-moonbeam`,
  },
  [CHAIN_IDS.MOONBEAM_TESTNET]: {
    domain: 'moonscan.io',
    subdomain: `${defaultEtherscanSubdomainPrefix}-moonbase`,
  },
  [CHAIN_IDS.MOONRIVER]: {
    domain: 'moonscan.io',
    subdomain: `${defaultEtherscanSubdomainPrefix}-moonriver`,
  },
  [CHAIN_IDS.GNOSIS]: {
    domain: 'gnosisscan.io',
    subdomain: `${defaultEtherscanSubdomainPrefix}-gnosis`,
  },
};
```

Is this the correct source?